lib/src/utils.dart

```diff
@@ -17,23 +17,19 @@ extension NullableStringUtils on String? {
 
 extension ValueUtils on Value {
   dynamic toPrimitive() {
-    if (hasNullValue()) {
-      return nullValue;
-    }
     if (hasNumberValue()) {
       return numberValue;
-    }
-    if (hasStringValue()) {
+    } else if (hasStringValue()) {
       return stringValue;
-    }
-    if (hasBoolValue()) {
+    } else if (hasBoolValue()) {
       return boolValue;
-    }
-    if (hasStructValue()) {
+    } else if (hasStructValue()) {
       return structValue.fields.map((key, value) => MapEntry(key, value.toPrimitive()));
-    }
-    if (hasListValue()) {
+    } else if (hasListValue()) {
       return listValue.values.map((e) => e.toPrimitive());
+    } else {
+      // default to null if no value found
```

Comment:
@njooma I realized if there was no value we wouldn't return anything, do we want to default to null?

Reply:
It should always have one of those fields set. So I don’t think null is necessarily bad and it is defensive, but I feel like an error might also make sense because it should be impossible to get to that state. But null is fine probs

Reply:
I think you're right, throwing an error makes a lot of sense. I will rework this.